cmd/branchingconfigmanagers/schedule-generator/main.go

```diff
@@ -0,0 +1,138 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io/fs"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/sirupsen/logrus"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/test-infra/prow/flagutil"
+	"k8s.io/test-infra/prow/logrusutil"
+	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+	crcontrollerutil "sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/yaml"
+
+	"github.com/openshift/ci-tools/pkg/api/ocplifecycle"
+)
+
+type options struct {
+	logLevel          string
+	scheduleDir       string
+	dryRun            bool
+	kubernetesOptions flagutil.KubernetesOptions
+}
+
+func gatherOptions() (*options, error) {
+	o := &options{kubernetesOptions: flagutil.KubernetesOptions{NOInClusterConfigDefault: true}}
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	fs.StringVar(&o.logLevel, "log-level", logrus.InfoLevel.String(), "Level at which to log output.")
+	fs.StringVar(&o.scheduleDir, "schedule-dir", "", "Directory holding schedules.")
+	o.kubernetesOptions.AddFlags(fs)
+	fs.BoolVar(&o.dryRun, "dry-run", true, "Do not mutate cluster state.")
+	if err := fs.Parse(os.Args[1:]); err != nil {
+		return o, fmt.Errorf("failed to parse flags: %w", err)
+	}
+	return o, nil
+}
+
+func validateOptions(o *options) error {
+	_, err := logrus.ParseLevel(o.logLevel)
+	if err != nil {
+		return fmt.Errorf("invalid --log-level: %w", err)
+	}
+	return o.kubernetesOptions.Validate(o.dryRun)
+}
+
+func main() {
+	logrusutil.ComponentInit()
+	o, err := gatherOptions()
+	if err != nil {
+		logrus.WithError(err).Fatal("Failed to gather options")
+	}
+	if err := validateOptions(o); err != nil {
+		logrus.WithError(err).Fatalf("Invalid options")
+	}
+	level, _ := logrus.ParseLevel(o.logLevel)
```

Comment:
lol was that error check too much typing :trollface:

Reply:
It's checked in the validation